### ISSUE TITLE
[BUG]  `npm completion` throws `This command does not support workspaces.`

### ISSUE THREAD

**Issue by b-fett** [npm/cli#5486]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Steps To Reproduce

* change working directory to some workspace
* run `npm completion`
* observe error message

### Environment

* `npm -v`: 8.19.1
* `node -v`: v16.15.0

**Comment by GerkinDev** [comment]
This has been the case for quite some time. Minimal repro here: https://github.com/GerkinDev/npm-completion-error

This makes npm very poorly usable in workspaces configuration in _plumbing mode_, at least for zsh. The issue can be _somewhat_ mitigated in zsh by disabling plumbing mode & statically loading completion with something like:

```sh
source <(cd $HOME && npm completion)
```

Doing the above in the profile file effectively disable completion entirely in workspace packages, but at least, the error is gone.

> Note: the `cd` ensure that you're not in a workspace package before running the command.

Unfortunately, with nvm, this becomes way less interesting.

Hope this gets fixed.

---

Full error & keywords (for people using Google like I did without finding anything for so long):

```
npm ERR! code ENOWORKSPACES
npm ERR! This command does not support workspaces.
```

zsh completion workspace

**Comment by dorianclay** [comment]
Was this fixed by this **fix: ignore implicit workspaces for completion** ([#5932](https://github.com/npm/cli/pull/5932))?

**Comment by lukekarrys** [comment]
I can confirm this was fixed by #5932.
